### PR TITLE
npm: Add no_bin_links option

### LINKED ISSUE #2128: Enable no-bin-links for community.general.npm
### Summary

I'm working on getting a vagrant development environment setup and am running into an issue where npm is having trouble installing on a synced folder. A solution would be to enable a flag to add the --no-bin-links setting to disable symlinking executables but there's no way to set this currently.

### Issue Type

Feature Idea

### Component Name

community.general.npm

### Additional Information

```yaml (paste below)
- name: NPM install for Login Service
  community.general.npm:
    path: "{{ project_root }}"
    links: no
  become_user: vagrant
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by aminvakil**
##### SUMMARY
Fixes #2128.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
npm

### CODE REVIEW

**Comment by aminvakil** [comment]
I think failing test is not relevant to this PR:
https://dev.azure.com/ansible/community.general/_build/results?buildId=10414&view=logs&j=829d0801-1ebf-5b85-0b62-ebcfaaac2311&t=5772b993-4b58-5c2f-94c2-569fd389dba1&l=716

**Comment by aminvakil** [comment]
Please do not merge this, I'll try to add integration test for this PR in the next few days.

**Comment by aminvakil** [comment]
rebuild_failed

**Comment by aminvakil** [comment]
Sanity tests probably get fixed with https://github.com/ansible/ansible/commit/4e2da0f89454ec46dc5aaf513a49b92e4a9a6052.

File: tests/integration/targets/npm/tasks/no_bin_links.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+- name: 'Remove any node modules'
+  file:
+    path: '{{ remote_dir }}/node_modules'
+    state: absent
+
+- vars:
+    # sample: node-v8.2.0-linux-x64.tar.xz
+    node_path: '{{ remote_dir }}/{{ nodejs_path }}/bin'
+    package: 'thelounge'
```
Comment (aminvakil) [inline tests/integration/targets/npm/tasks/no_bin_links.yml:10]: I should probably change this package to a simpler package which creates `.bin` folder, but for now this works :)

Comment (felixfontein) [reply tests/integration/targets/npm/tasks/no_bin_links.yml:10]: Yeah, that would be great. I searched a bit on npm, and found https://www.npmjs.com/package/ncp which has no dependencies and installs a CLI program.

Comment (aminvakil) [reply tests/integration/targets/npm/tasks/no_bin_links.yml]: Thank you very much for your suggestion, it fixed all tests as well.

**Comment by aminvakil** [comment]
ready_for_review

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@skluthe can you try out this PR whether it works for you? Thanks!

**Comment by felixfontein** [comment]
@aminvakil thanks for implementing this!

**Pull request merged**